### DISCUSSION TITLE
Run terragrunt run-all in setup with find_in_parent_folders

### DISCUSSION THREAD

**Discussion by simonszu** [gruntwork-io/knowledge-base#815] [How do I]
I have a couple AWS environments which are built up similarly and managed by terragrunt. I have a directory structure where i have a folder called `environments` which contains a `terragrunt.hcl` with state config, etc. 
Inside the environments folder is a subfolder per environment, each subfolder contains a terragrunt.hcl which includes the root file from the environments folder with the help of 
```
include {
    path = find_in_parent_folders()
}
```

Now, i want to fully use the superpowers of terragrunt and execute the `run-all` command. This causes some problems: When i execute it in the `environments` root folder, it properly cycles through all the environments subfolder, however it also tries to execute the root `terragrunt.hcl` which was never meant to be executed but only included in the subfolders. 

How can i run `terragrunt run-all` in such a way that it recognizes the root hcl file to not being "executable" on its own, but has to be included in the environment subfolder? 

---

<ins datetime="2024-04-10T09:43:09Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110872">Tracked in ticket #110872</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
can be used multiple approaches:
* pass `--terragrunt-exclude-dir`:
```
terragrunt run-all apply --terragrunt-exclude-dir $(pwd)
```
* pass `--terragrunt-strict-include` but it will require to pass individual directory to include:
```
terragrunt run-all apply --terragrunt-strict-include --terragrunt-include-dir env1 --terragrunt-include-dir env2 ...
```
* can be defined `skip = true` in the parent file - it will still appear in module list, but be skipped during execution...
```
skip = true
```

https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-exclude-dir

Simplified example in: https://github.com/denis256/terragrunt-tests/tree/master/discussion-815

**Comment by yhakbar** [comment]
FWI, there's a simple recommendation that we make nowadays to just call the root configuration something else that resolves this issue:
https://terragrunt.gruntwork.io/docs/migrate/migrating-from-root-terragrunt-hcl/
